src/sentry/sentry_metrics/configuration.py

```diff
@@ -19,14 +19,9 @@ def from_str(use_case: str) -> "UseCaseKey":
             raise ValueError
 
 
-class DbKey(Enum):
-    STRING_INDEXER = "StringIndexer"
-    PERF_STRING_INDEXER = "PerfStringIndexer"
-
-
 @dataclass(frozen=True)
 class MetricsIngestConfiguration:
-    db_model: DbKey
+    db_backend: str
```

Comment:
I think the `MetricsIngestConfiguration` we designed is not sufficient for this use case and we need to rethink it a bit to do this task in a safe way.

Context on `MetricsIngestConfiguraiton`.
When we introduced the performance indexer (and we managed to break release health) it became clear fairly quickly that there was a number of parameters that had to be specific to each indexer and that we should group them under some structure that would represent an indexer instance to remain sane. Otherwise we would have had a number of independent configurations that we would have had to override. With that solution we were pretty much sure to make mistakes and to cause production issues.
Thus the `metricsIngestConfiguration` was born. The idea was that anywhere you needed an indexer you would reference it through the `use_case_id`, every indexer public method requires a `use_case_id` with no default, so no mistakes can be made and the indexer never reference settings unless through this object. 

The `use_case_id`, thus, is meant to represent the product concept (release health and performance so far). Nothing more.  This is because the product code using the indexer is generally product specific and should not change if we make changes to the indexer storage.

Now this use case (two or more indexers for performance) breaks everything.
I think we may even (by the end of next week) end up running multiple cloud spanner indexers in parallel in prod with different transactions models. 

This PR seems to be overriding specific fields in the `metricsIngestConfiguration` depending on which db_backend we want when we call `get_ingest_config`. 
I think we should instead provide a totally different `MetricsIngestConfiguration` object depending on the db_backend. This would make it easier to customize all its field and provide a more detailed DB configuration. 
See below for more details on which additional fields we should have here.